cask-repair

```diff
@@ -561,12 +561,23 @@ for cask in "${@}"; do
 
   git commit "${cask_file}" --message "${commit_message}" --quiet
   git push --force "${cask_repair_remote_name}" "${cask_branch}" --quiet 2> "${submission_error_log}"
+  gitpush_result="${?}"
```

Comment:
i need the `gitpush_result` otherwise the `${?}` only hold the last result of previous command and i can not check the result in an second `if` in line 574

Reply:
Both times you’re saving the result of the command and then checking it right away. This variable is useless.

Reply:
not at all useless

the idea is to have 2 possible pull requests which can fail and only one check for it. thats why i set the `gitpush_result` 2 times and also the fill again the log with `submission_error_log`

if i check it with `${?}` in line 567 then the result is gone and i can not check the result 563 in line 574.

i like to check it also in the case when the first `if` is not true

Reply:
when i do it like this:

```
  git push --force "${cask_repair_remote_name}" "${cask_branch}" --quiet 2> "${submission_error_log}"

  if [[ "${?}" -ne 0 ]] && grep --quiet 'shallow update not allowed' "${submission_error_log}"; then
    echo 'Push failed due to shallow repo. Unshallowing…'
    HOMEBREW_NO_AUTO_UPDATE=1 brew tap --full "caskroom/$(current_tap)"
    git push --force "${cask_repair_remote_name}" "${cask_branch}" --quiet 2> "${submission_error_log}"
  fi

  if [[ "${?}" -ne 0 ]]; then 
```
then the second `if [[ "${?}"` not catch the error.

Reply:
That’s not what happens in my test:

```
if [[ 'a' == 'a' ]]; then # Just so it continues
  curl 'fake data' 2> /dev/null # This produces an error
fi

echo "${?}" # It correctly outputs a `6`, the result of the `curl` command.

Reply:
thats correct. for this case we don't need an variable.


this is when the first `if` is true. 

```
curl 'fake data' 2> /dev/null # This produces an error nr 1

if [[ "${?}" -ne 0 ]] && [[ 'b' == 'a' ]]; then # this is not true because b!=a
  curl 'fake data' 2> /dev/null # This produces an error nr 2 - but is never executed
fi

if [[ "${?}" -ne 0 ]]; then
  # this is never called because here is it always 0 and not the result from error 1
if
```

i need to check it if curl 1 makes an error but curl 2 never exected

Reply:
I see what you mean. I’d still rather deal with it a different way. Remove the `gitpush_result` and make it a nested `if`:

```
if [[ "${?}" -ne 0 ]]; then
  if grep --quiet 'shallow update not allowed' "${submission_error_log}"; then
    …
  else
    …
  fi
fi
```

This way we have a clearer section to deal with other errors in the future, instead of keeping overwriting the same variable.